lib/client.ts

```diff
@@ -1,13 +1,303 @@
+        contentEncoding: ContentEncoding,
+        options?: IngestOptions,
+    ): Promise<IngestStatus> =>
+        this.client
+            .post<IngestStatus>(this.localPath + '/' + id + '/ingest', stream, {
+                headers: {
+                    'Content-Type': contentType,
+                    'Content-Encoding': contentEncoding,
+                },
+                params: {
+                    'timestamp-field': options?.timestampField,
+                    'timestamp-format': options?.timestampFormat,
+                    'csv-delimiter': options?.csvDelimiter,
+                },
+            })
+            .then((response) => {
+                return response.data;
+            });
+
+    ingestBuffer = (
+        id: string,
+        buffer: Buffer,
+        contentType: ContentType,
+        contentEncoding: ContentEncoding,
+        options?: IngestOptions,
+    ): Promise<IngestStatus> => this.ingest(id, Readable.from(buffer), contentType, contentEncoding, options);
+
+    ingestString = (
+        id: string,
+        data: string,
+        contentType: ContentType,
+        contentEncoding: ContentEncoding,
+        options?: IngestOptions,
+    ): Promise<IngestStatus> => this.ingest(id, Readable.from(data), contentType, contentEncoding, options);
+
+    ingestEvents = async (
+        id: string,
+        events: Array<object> | object,
+        options?: IngestOptions,
+    ): Promise<IngestStatus> => {
+        const array = Array.isArray(events) ? events : [events];
+        const json = array.map((v) => JSON.stringify(v)).join('\n');
+        const encoded = await promisify(gzip)(json);
+        return this.ingestBuffer(id, encoded, ContentType.NDJSON, ContentEncoding.GZIP, options);
+    };
+
+    queryLegacy = (id: string, query: QueryLegacy, options?: QueryOptions): Promise<QueryLegacyResult> =>
+        this.client
+            .post<QueryLegacyResult>(this.localPath + '/' + id + '/query', query, {
+                params: {
+                    'streaming-duration': options?.streamingDuration,
+                    nocache: options?.noCache,
+                },
+            })
+            .then((response) => {
+                return response.data;
+            });
+
+    query = (apl: string, options?: QueryOptions): Promise<QueryResult> => {
+        const req: Query = { apl: apl, startTime: options?.startTime, endTime: options?.endTime };
+        return this.client
+            .post<QueryResult>(this.localPath + '/_apl', req, {
+                params: {
+                    'streaming-duration': options?.streamingDuration,
+                    nocache: options?.noCache,
+                    format: 'legacy',
+                },
+            })
+            .then((response) => {
+                return response.data;
+            });
+    };
+
+    aplQuery = (apl: string, options?: QueryOptions): Promise<QueryResult> => this.query(apl, options);
+}
+
+export enum ContentType {
```

Comment:
I think we can keep the models in the datasets file, wdyt?

Reply:
hmm I am not sure about that, I feel its a little bit weird to use `client.ingest` and then use params from a different namespace. how is it in the rs sdk?

Reply:
after our talk, I believe this one is ready to be merged.